cdi/src/main/java/io/smallrye/context/inject/SmallryeContextCdiExtension.java

```diff
@@ -371,4 +373,41 @@ public void cleanup(@Observes AfterDeploymentValidation adv) {
         this.executorMap.clear();
         this.threadContextMap.clear();
     }
+
+    // Used to delay the ConfigProvider.getConfig() call until we need the config
+    // Doing it too early may break application deployment in application servers
+    @Alternative
```

Comment:
If I don't add this I get the following error when trying to inject the Config into a test class:
```
    Caused by: org.jboss.weld.exceptions.DeploymentException: WELD-001409: Ambiguous dependencies for type Config with qualifiers @Default
  at injection point [BackedAnnotatedField] @Inject org.wildfly.test.integration.microprofile.context.propagation.ContextPropagationSanityTestCase.config
  at org.wildfly.test.integration.microprofile.context.propagation.ContextPropagationSanityTestCase.config(ContextPropagationSanityTestCase.java:0)
  Possible dependencies: 
  - Managed Bean [class io.smallrye.context.inject.SmallryeContextCdiExtension$LazyConfig] with qualifiers [@Any @Default],
  - Producer Method [Config] with qualifiers [@Any @Default] declared as [[UnbackedAnnotatedMethod] @Produces io.smallrye.config.inject.ConfigProducer.getConfig(InjectionPoint)]
"}}}}
```
Is this the proper way to 'disable' a bean for injection? Or is there a better way?

Reply:
@kabir sorry for late response, been gone over xmass period.
Alternative works but it isn't really for disabling beans. I'd suggest you either make the class non-static in which case CDI shouldn't pick it up ([inner classes are not managed beans](https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#what_classes_are_beans)) or, if you want to keep it like this, then use `@Vetoed` annotation instead of `@Alternative`.

Reply:
@manovotn No problem :) I made the class non-static